### DISCUSSION TITLE
Canvas other UTF-8/16/32 character support

### DISCUSSION THREAD

**Discussion by Kolky** [spectreconsole/spectre.console#1704] [Ideas]
I've wrote another version of the `Canvas` that supports other rendering characters than the default space (" "). It supports providing UTF-8, 16 and 32 characters. When another character is seen than the default space, it will use the foreground color instead of the background color. This allows you to use the `Canvas` with blocks or other special characters.

Few things before I make a PR:
* Should I integrate it into the existing `Canvas`? Or keep it a separate class?
* If so, what is a good name?
* I now use reflection, with integer for UTF-32, char for UTF-8 or UTF-16 and string for all. But can also change it to use overloads.

**Comment by phil-scott-78** [comment]
Can't comment on anything besides the last bullet point - avoid reflection. With apt/trimming enabled now, you'll get warnings and a mess of headaches. Being explicit about the types is what you'll want
